sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/StorageChannelMode.java

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.common;
+
+/**
+ * Mode to open a Storage-backed SeekableByteChannel.
+ */
+public enum StorageChannelMode {
```

Comment:
Do we want to provide a separate type for this or constrict which `StandardOpenOptions` could be used?

Reply:
I'm not entirely sure because I'm not sure what to do with the distinction of how overwrite toggling works across different resource types. It may be that we actually create a different enum per-resource-type since they all behave slightly differently
- block blobs can theoretically at most through this interface overwrite, append, and prepend
- append blobs can overwrite or append
- page blobs can overwrite or random access write (so long as it writes to offsets of 512 multiples)
- datalake file can probably at most overwrite or append
- share files can overwrite or random access write

There's several nuances here and I'm not sure that can all be captured in a simple read/write plus an overwrite flag without users making assumptions about what overwrite=false actually means in this context.

Reply:
And this assumes that opening the channel will take care of the create/delete/if-exists behavior automatically. If we require a caller assure state of the resource before opening the channel, what we need to account for changes. I'm still working on exactly how we expect a user to work with this.